src/RouteProvider/AbstractEntityRouteProvider.php

```diff
@@ -71,8 +72,15 @@ private function processValidSubscriptions(Subscriptions $subscriptions, array $
                 continue;
             }
 
-            if (isset($subscription['if']) && false === $this->getExpressionLanguage()->evaluate($subscription['if'], ['obj' => $entity])) {
-                continue;
+            if (isset($subscription['if'])) {
+                $result = $this->getExpressionLanguage()->evaluate($subscription['if'], ['obj' => $entity]);
+                if (!\is_bool($result)) {
+                    throw new InvalidIfResultException($subscription['if'], $result);
+                }
+
+                if (false === $result) {
```

Comment:
@Brajk19 Since we know `$result` is a bool, why not:

```suggestion
                if (!$result) {
```